test/e2e/util.go

```diff
@@ -54,6 +56,27 @@ func waitForPodRunning(c *client.Client, id string) {
 	}
 }
 
+// waitForPodNotPending returns false if it took too long for the pod to go out of pending state.
+func waitForPodNotPending(c *client.Client, podName string) bool {
+	for i := 0; i < 10; i++ {
+		if i > 0 {
+			time.Sleep(5 * time.Second)
+		}
+		pod, err := c.Pods(api.NamespaceDefault).Get(podName)
+		if err != nil {
+			glog.Warningf("Get pod %s failed: %v", podName, err)
+			continue
+		}
+		if pod.Status.Phase != api.PodPending {
+			glog.Infof("Saw pod %s out of pending state (found %q)", podName, pod.Status.Phase)
+			return true
+		}
+		glog.Infof("Waiting for pod %s status to be !%q (found %q)", podName, api.PodPending, pod.Status.Phase)
+	}
+	glog.Warningf("Gave up waiting for pod %s status to go out of pending", podName)
```

Comment:
Good idea, but we should do this in a separate PR for improvements to the
util.go file.

S

On Fri, Jan 23, 2015 at 5:26 PM, Dawn Chen <notifications@github.com> wrote:

> In test/e2e/util.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3780#discussion_r23492243>
> :
>
> > +	for i := 0; i < 10; i++ {
> > +		if i > 0 {
> > +			time.Sleep(5 * time.Second)
> > +		}
> > +		pod, err := c.Pods(api.NamespaceDefault).Get(podName)
> > +		if err != nil {
> > +			glog.Warningf("Get pod %s failed: %v", podName, err)
> > +			continue
> > +		}
> > +		if pod.Status.Phase != api.PodPending {
> > +			glog.Infof("Saw pod %s out of pending state (found %q)", podName, pod.Status.Phase)
> > +			return true
> > +		}
> > +		glog.Infof("Waiting for pod %s status to be !%q (found %q)", podName, api.PodPending, pod.Status.Phase)
> > +	}
> > +	glog.Warningf("Gave up waiting for pod %s status to go out of pending", podName)
>
> When we give up on waiting or eventual bailout, how about dumping
> pod.Status for debugging?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3780/files#r23492243>
> .
>